ledger/appcow.go

```diff
@@ -519,6 +532,10 @@ func (cb *roundCowState) BuildEvalDelta(aidx basics.AppIndex, txn *transactions.
 			}
 		}
 	}
+
+	// logs
+	evalDelta.Logs = cb.logs
```

Comment:
I think the intent is to have `LogItem.ID` refer to an index into `txn.ForeignApps` for log items when they are in the `EvalDelta` struct, since that will save space compared to storing a larger app ID for every log item.

The question is what should `LogItem.ID` refer to inside of `LedgerForLogic`/`roundCowState`?

Currently it refers to the real app ID. If we want to keep it that way, we'll need to convert each log item's ID into an index into `txn.ForeignApps`. That should happen in this function, since it also converts local state delta addresses into indexes into `txn.Accounts` (the same should also happen in the `GetDelta` function from `data/transactions/logic/evalStateful_test.go`). Though if we decide to do this, I'd recommend creating two structs to represent the two different `LogItem` versions: one with app ID and one with index into `txn.ForeignApps`. Otherwise it'll be difficult to keep track what `ID` actually means.

The alternative would be to make `LogItem.ID` always be an index into `txn.ForeignApps`. Then the `AppendLog` functions could convert the app ID to the index when the `LogItem` is created.

Reply:
I would choose the alternative because I would prefer not to keep track of different LogItem versions.